src/main/bash/sdkman-main.sh

```diff
@@ -153,8 +153,8 @@ function sdk() {
 	fi
 
 	# Attempt upgrade after all is done
-	if [[ "$COMMAND" != "selfupdate" ]]; then
-		__sdkman_auto_update "$SDKMAN_REMOTE_VERSION" "$SDKMAN_VERSION"
+	if [[ "$COMMAND" != "selfupdate" && "$sdkman_selfupdate_enable" != "false" ]]; then
```

Comment:
This double negative seems strange. Why not `sdkman_selfupdate_enable == "true"`?

Reply:
I'm 100% against assuming everyone will have this set in their config files. For example, I manage my dotfiles using another program, so I'm the one who makes changes to it. If later in the future sdkman introduces new config variables, I would miss it. 

I think 2 things are different:

1. Default settings 
2. User preferences (the ~/.sdkman/etc/config file)

And I don't think sdkman should make changes unattended(ly) to the user preferences, but only to the default settings. Just like Git (~/.gitconfig) or any other tool.

And one way to accomplish this behavior is doing such negative comparison:

It will only work if the user setted to false, explicitly, otherwise it will have the truthy behavior (default=true).

Reply:
This is a huge leap from how SDKMAN currently works, and I don't think we should shoehorn it into this PR. If we want to go down this route, it should be done consistently for all the configurations, and it should be done as a separate PR.

For this PR, I expect it to be done as it always was. I suggest you open a new feature request for the above.

Reply:
Cool. That works for me.